packages/dart_code_builder/src/example.candy

```diff
@@ -0,0 +1,140 @@
+use ..nodes
+
+fun main() {
+  let list = MutableList.empty<Directive | Declaration>()
+  list.append(Import(
+    "package:foo/foo.dart",
+    alias = Some<String>("foo"),
+    show = List.empty<String>(),
+    hide = List.of1<String>("bar")
+  ))
+  list.append(Import(
+    "dart:blub",
+    alias = None<String>(),
+    show = List.of1<String>("blubber"),
+    hide = List.empty<String>(),
+  ))
+  list.append(Part("test.dart"))
+  list.append(PartOf("meow.dart"))
+  list.append(Function(
+    "main",
+    annotations = List.of1<Annotation>(
+      Annotation(
+        (UserType(
+          "Cached",
+          parameters = List.empty<DartType>(),
+          prefix = None<String>(),
+        ) as Expression).call(
+          positionalArguments = List.empty<Expression>(),
+          namedArguments = Map.empty<String, Expression>(),
+          typeArguments = List.empty<DartType>(),
+        )
+      )
+    ),
+    _returns = Some<DartType>(UserType(
+      "Foo",
+      parameters = List.empty<DartType>(),
+      prefix = Some<String>("foo"),
+    )),
+    requiredParameters = List.of1<Parameter>(
+      Parameter(
+        "args",
+        type = Some<DartType>(UserType(
+          "List",
+          parameters = List.of1<DartType>(UserType(
+            "String",
+            parameters = List.empty<DartType>(),
+            prefix = None<String>(),
+          )),
+          prefix = None<String>(),
+        )),
+        defaultValue = None<Expression>(),
+      ),
+    ),
+    positionalParameters = List.of1<Parameter>(
+      Parameter(
+        "number",
+        type = Some<DartType>(UserType("int", List.empty<DartType>(), None<String>())),
+        default = Some<Expression>(Literal(123)),
+      ),
+    ),
+    namedParameters = List.empty<Parameter>(),
+    body = None<Body>(),
+  ))
+  list.append(Class(
+    "Foo",
+    annotations = List.of1<Annotation>(Annotation(Identifier("data", prefix = None<String>()))),
+    extends_ = Some<DartType>(UserType("Bar", List.empty<DartType>(), None<String>())),
+    implements_ = List.of3<DartType>(
+      UserType("Baz", List.empty<DartType>(), None<String>()),
+      UserType("Wheez", List.empty<DartType>(), None<String>()),
+      UserType("Fnop", List.empty<DartType>(), None<String>()),
+    ),
+    with_ = List.of2<DartType>(
+      UserType("Whup", List.empty<DartType>(), None<String>()),
+      UserType("Flang", List.empty<DartType>(), None<String>()),
+    ),
+    body = List.of4<Field | Getter | Setter | Function>(
+      // static const numBlubs = 123;
+      Field("numBlubs", isStatic = true, mutability = Const(), Some<Expression>(Literal(123))),
```

Comment:
To my knowledge, only the following modifier combinations are valid:
* \<nothing\>
* `final`
* `late`
* `late final`
* `external`
* `external static`
* `external covariant`
* `covariant`
* `covariant late`
* `static`
* `static final`
* `static const`
* `static late`
* `static late final`

https://pub.dev/documentation/analyzer/latest/dart_ast_ast/FieldDeclaration-class.html